src/annotator/config/index.js

```diff
@@ -56,62 +66,98 @@ function configurationKeys(appContext) {
 }
 
 /**
- * Reads the Hypothesis configuration from the environment.
- *
- * @param {string[]} settingsKeys - List of settings that should be returned.
- * @param {Window} window_ - The Window object to read config from.
+ * The definition for configuration sources, their default values
+ * @return {ConfigDefinitionMap}
  */
-function configFrom(settingsKeys, window_) {
-  const settings = settingsFrom(window_);
-  const allConfigSettings = {
-    annotations: settings.annotations,
+function configDefinitions(settings) {
+  return {
+    annotations: {
+      valueFn: () => settings.annotations,
+    },
+    appType: {
+      valueFn: () =>
+        settings.hostPageSetting('appType', {
+          allowInBrowserExt: true,
+        }),
+    },
     // URL where client assets are served from. Used when injecting the client
     // into child iframes.
-    assetRoot: settings.hostPageSetting('assetRoot', {
-      allowInBrowserExt: true,
-    }),
-    branding: settings.hostPageSetting('branding'),
+    assetRoot: {
+      valueFn: () =>
+        settings.hostPageSetting('assetRoot', {
+          allowInBrowserExt: true,
+        }),
+    },
+    branding: {
+      valueFn: () => settings.hostPageSetting('branding'),
+    },
     // URL of the client's boot script. Used when injecting the client into
     // child iframes.
-    clientUrl: settings.clientUrl,
-    enableExperimentalNewNoteButton: settings.hostPageSetting(
-      'enableExperimentalNewNoteButton'
-    ),
-    experimental: settings.hostPageSetting('experimental', {
-      defaultValue: {},
-    }),
-    group: settings.group,
-    focus: settings.hostPageSetting('focus'),
-    theme: settings.hostPageSetting('theme'),
-    usernameUrl: settings.hostPageSetting('usernameUrl'),
-    onLayoutChange: settings.hostPageSetting('onLayoutChange'),
-    openSidebar: settings.hostPageSetting('openSidebar', {
-      allowInBrowserExt: true,
-      // Coerce value to a boolean because it may come from via as a string
-      coerce: toBoolean,
-    }),
-    query: settings.query,
-    requestConfigFromFrame: settings.hostPageSetting('requestConfigFromFrame'),
-    services: settings.hostPageSetting('services'),
-    showHighlights: settings.showHighlights,
-    notebookAppUrl: settings.notebookAppUrl,
-    sidebarAppUrl: settings.sidebarAppUrl,
-    // Subframe identifier given when a frame is being embedded into
+    clientUrl: {
+      valueFn: () => settings.clientUrl,
+    },
+    enableExperimentalNewNoteButton: {
+      valueFn: () =>
+        settings.hostPageSetting('enableExperimentalNewNoteButton'),
+    },
+    experimental: {
+      valueFn: () =>
+        settings.hostPageSetting('experimental', {
```

Comment:
Again, this has no app context and is not used anywhere in our client as near as I can tell. Should we add this to the sidebar context?